integrations/tensorflow/bindings/python/pyiree/tf/support/tf_test_utils.py

```diff
@@ -427,6 +446,22 @@ def serialize(self, trace_dir: str) -> None:
       call_dir = os.path.join(trace_dir, f"call_{str(i).zfill(width)}")
       call.serialize(call_dir)
 
+    # C++ Serialization.
+    if "tf" not in self.backend:
```

Comment:
This feels pretty gross. Do we have any way to delegate to the CompiledModule class hierarchy to handle this switch?

Reply:
Agreed. I added a flag onto the Trace to directly check this.

Reply:
Ah I was thinking something more like having a method on CompiledModule like `supports_cxx_serialization` rather than the trace having to know about TF vs not-TF

Reply:
The trace doesn't have access to the `CompiledModule` after initialization, but I think that's a better name for the flag than `backend_is_tf`.

Reply:
At initialization you can ask the CompiledModule whether it supports C++ serialization though rather than introspecting on the class hierarchy. Then traces don't have to know about the class hierarchy.

Reply:
Done.